### DISCUSSION TITLE
Undefined symbol __atomic_load_8 error when downloading spacy models on RPI 4

### DISCUSSION THREAD

**Discussion by InputMoe** [explosion/spaCy#7372] [Help: Installation] [answered]
When I install the module on Raspberry Pi 4 `Linux raspberrypi 5.4.79-v7l+ #1373 SMP Mon Nov 23 13:27:40 GMT 2020 armv7l GNU/Linux` it ended up with the same error multiple times:

`undefined symbol __atomic_load_8`

when downloading a model with

`python -m spacy download ...`

* Readme.md
  * Step a) ends with `undefined symbol __atomic_load_8`
  * Step b)
   * compiling with `cortexa57` and `cortexa53` ended with a gcc error: `armv8-a does not support feature 'fp'`
   * compiling with `cortexa15` and `cortex9` succeeded but also ran in the `atomic_load_8` error when downloading a model
 
* stackoverflow (https://stackoverflow.com/questions/59927844/is-it-possible-to-install-spacy-to-raspberry-pi-4-raspbian-buster)
  * compiling with `cortexa57 and `cortexa53` ended with a gcc error: 'armv8-a does not support feature 'fp', did you mean +nofp'
  * compiling with `cortexa15` and `cortexa9` succeeded but also ran in the `atomic_load_8` error when downloading a model

Are there other ways for a proper installation?

Add: the docker option does not work, because the container is not compatible with the arm architecture

**Comment by adrianeboyd** [comment, marked as answer]
I'm not sure exactly what leads to these particular errors, but you have a 32-bit distro (`armv7l` indicates that it's 32-bit) and you will need a 64-bit OS + 64-bit python for spaCy v2 to work correctly. See the notes at the top of this comment for installing / checking that you have a 64-bit OS:

https://github.com/explosion/spaCy/issues/6021#issuecomment-687059113

The install instructions in that comment are now a bit out-of-date, too. We recently improved the source distributions for our packages so that the build dependencies are correct and so that `blis` will choose `cortexa57` by default for linux+aarch64, so as of spacy v2.3.5 you should now be able to install spacy in a venv with only:

```bash
python3 -m venv .venv
source .venv/bin/activate
pip install --upgrade pip setuptools
pip install spacy
```

It will take ~20 minutes to compile everything.

You can run the test suite after installing a few additional packages:

```bash
pip install pytest pytest-timeout mock jsonschema pyrsistent
pytest --pyargs spacy
```

The test suite will take ~5 minutes to run.

Alternatively, there are binary linux aarch64 packages on conda-forge. There's a linux aarch64 miniforge installer and the install command would be `conda install spacy`. If you're not using the miniforge installer, then you need to add `-c conda-forge` or otherwise add the conda-forge channel.

**Comment by InputMoe** [comment]
yes this worked!

**Comment by InputMoe** [comment]
thx!
